catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -769,6 +790,18 @@ private void correctAllNestedReferences(Script script) {
 				ifBeginList.remove(ifBeginBrick);
 			}
 		}
+
+		removeBricksWithInvalidReferences(bricksWithInvalidReferences, script);
+	}
+
+	private void removeBricksWithInvalidReferences(ArrayList<Brick> bricksWithInvalidReferences, Script script) {
+		for (Brick brick : bricksWithInvalidReferences) {
+			script.removeBrick(brick);
+		}
+	}
```

Comment:
I think the script itself is a better place for this method. `script.removeBricks(List bricks)`. Then you can use `script.removeBricks(bricksWithInvalidReferences)`.